src/github.com/matrix-org/dendrite/roomserver/storage/sql.go

```diff
@@ -326,20 +363,46 @@ func (s *statements) prepareEvents(db *sql.DB) (err error) {
 	if s.insertEventStmt, err = db.Prepare(insertEventSQL); err != nil {
 		return
 	}
+	if s.selectStateEventsByIDStmt, err = db.Prepare(selectStateEventsByIDSQL); err != nil {
+		return
+	}
 	return
 }
 
 func (s *statements) insertEvent(
 	roomNID, eventTypeNID, eventStateKeyNID int64,
 	eventID string,
 	referenceSHA256 []byte,
+	authEventNIDs []int64,
```

Comment:
It might, but I worry than I'd run into sadness when I wanted to pass slices of NIDs into database functions. (Currently I can just cast the slices to Int64Array and postgres will do the rest, but I can't cast []NID to []int64.)